dolphin/alert_manager/trap_receiver.py

```diff
@@ -109,25 +130,42 @@ def _cb_fun(self, state_reference, context_engine_id, context_name,
                      context_name.prettyPrint(), exec_context['securityModel'],
                      exec_context['securityName']))
 
-        var_binds = [rfc1902.ObjectType(
-            rfc1902.ObjectIdentity(x[0]),
-            x[1]).resolveWithMib(self.mib_view_controller)
-            for x in var_binds]
-        alert = {}
-
-        for var_bind in var_binds:
-            oid, value = self._extract_oid_value(var_bind)
-            alert[oid] = value
-
-        # Fill additional info to alert_info
-        # transportAddress contains both ip and port, extract ip address
-        alert['transport_address'] = exec_context['transportAddress'][0]
-
-        # Handover trap info to alert processor for model
-        # translation and export
         try:
+            # transportAddress contains both ip and port, extract ip address
+            source_ip = exec_context['transportAddress'][0]
+            alert_source = self._get_alert_source_by_host(source_ip)
+
+            # In case of non v3 version, community string is used to map the
+            # trap. Pysnmp library helps to filter traps whose community string
+            # are not configured. But if a given community name x is configured
+            # for storage1, if the trap is received with x from storage 2,
+            # library will allow the trap. So for non v3 version, we need to
+            # verify that community name is configured at alert source db for
+            # the storage which is sending traps.
+            if exec_context['securityModel'] != constants.SNMP_V3_VERSION:
+                # context_name contains the incoming community string value
+                if alert_source['community_string'] != str(context_name):
```

Comment:
to avoid multipe if's, a single AND operation can be added.:
ex: if exec_context['securityModel'] != constants.SNMP_V3_VERSION **and** alert_source['community_string'] != str(context_name):

Reply:
Done